File: Leetcode/Weekly Contest 241/WC241-C.cpp
class FindSumPairs {
public:
	vector<int> n1;
	vector<int> n2;
	unordered_map<int, int> m2;
	FindSumPairs(vector<int>& nums1, vector<int>& nums2) {
		n1 = nums1;
		n2 = nums2;
		for (int i = 0; i < n2.size(); i++)
			m2[n2[i]]++;
	}

	void add(int index, int val) {
		m2[n2[index]]--;
		n2[index] += val;
		m2[n2[index]]++;
	}

	int count(int tot) {
		int ans = 0;
		for (int i = 0; i < n1.size(); i++)
		{
			int target = tot - n1[i];
			ans += m2[target];
		}
		return ans;
	}
};

File: Leetcode/Weekly Contest 241/WC241-C.py
class FindSumPairs:
    def __init__(self, nums1: List[int], nums2: List[int]):
        self.n1 = nums1
        self.n2 = nums2
        self.m = {}
        for x in nums2:
            if x in self.m:
                self.m[x] +=1
            else:
                self.m[x] = 1

    def add(self, index: int, val: int) -> None:
        self.m[self.n2[index]] -=1
        self.n2[index] += val
        
        if self.n2[index] in self.m:
            self.m[self.n2[index]] +=1
        else:
            self.m[self.n2[index]] = 1

    def count(self, tot: int) -> int:
        return sum([self.m[tot-x] if tot-x in self.m else 0 for x in self.n1])